src/cpp/session/modules/SessionTerminalShell.cpp

```diff
@@ -78,6 +78,7 @@ void scanAvailableShells(std::vector<TerminalShell>* pShells)
 {
 #ifdef _WIN32
    std::vector<std::string> args;
+   std::vector<std::string> bashLoginArgs = { "-l" };
```

Comment:
This is required as otherwise the shell isn't treated as a login shell, and the expected profile isn't sourced. (Note that the "default" shell profile _does_ include `-l` so this is specific to Windows bash discovery.)